### PR TITLE
check internal "rustc" attrs

### PR DESCRIPTION
**Pull request by tjhance**
The rust stdlib has a prevalance of internal "rustc" attributes, which vary greatly in how relevant they are to verification.

This PR includes an initial whitelist of "safe" rustc attributes, and errors if it encounters an attribute that it doesn't recognize. As we verify the stdlib, we'll likely come across some attributes that require special support. For example, `rustc_never_returns_null_ptr` and `rustc_nowind` will require additional proof obligations.

### CODE REVIEW

File: source/rust_verify/src/fn_call_to_vir.rs [hunk 1]
```diff
@@ -143,8 +143,10 @@ pub(crate) fn fn_call_to_vir<'tcx>(
     }
 
     let f_attrs = bctx.ctxt.tcx.get_attrs_unchecked(f);
-    let f_vattrs = get_verifier_attrs(f_attrs, Some(&mut *bctx.ctxt.diagnostics.borrow_mut()))?;
-    if f_vattrs.internal_get_field_many_variants {
+    if crate::attributes::is_get_field_many_variants(
+        f_attrs,
+        Some(&mut *bctx.ctxt.diagnostics.borrow_mut()),
+    )? {
         return Err(vir::messages::error(
```
Comment (utaal-b) [inline source/rust_verify/src/fn_call_to_vir.rs:150]: Can you explain why this change is necessary: i.e. why you cannot `get_verifier_attrs`  via `bctx.ctxt` here?

Comment (tjhance) [reply source/rust_verify/src/fn_call_to_vir.rs:150]: Sometimes it would examine a function in the rust standard lib that we support via external_fn_specification.

Comment (utaal-b) [reply source/rust_verify/src/fn_call_to_vir.rs:150]: Can you please add a comment to explain that?

**Review by utaal — APPROVED** [review]
